### ISSUE TITLE
Needs a better way to differentiate multiple clusters' metrics (like adding custom labels to metrics)

### ISSUE THREAD

**Issue by Frefreak** [open-telemetry/opentelemetry-collector#4716]
**Is your feature request related to a problem? Please describe.**
When we have multiple deployments/clusters with the same collector binary, and the metrics of them are all collected by the same prometheus, we needs a way to differentiate the metrics (e.g. we need to classify `otelcol_receiver_accepted_spans` by clusters). In some environments the prometheus are provided by another group/team/PaaS and we have no control of the config of it, it would be convenient if there's a way to let the otlp collector output metrics of different labels or namespaces when configured.

Before, one can use `--metrics-prefix` to achieve this, but it seems been removed in recent versions. I think moving the metrics config to yaml is definitely a good move, but it seems currently there's no easy way to achieve the same or similar effect as before.

As mentioned by the [changelog](https://github.com/open-telemetry/opentelemetry-collector/blob/main/CHANGELOG.md#:~:text=%2D%2Dmetrics%2Dprefix%20is%20no%20longer%20operative%3B%20the%20prefix%20is%20determined%20by%20the%20value%20of%20service.buildInfo.Command) currently we can use `service.buildInfo.Command` to configure  it, but it appears the value is not actually used by the init code: https://github.com/open-telemetry/opentelemetry-collector/blob/e080cae/service/telemetry.go#L162. And even if this got fixed, since it is a compile time value it would still not be very flexible.

**Describe the solution you'd like**
As the metrics config now can be configured by `service.telemetry.metrics`, ideally I would hope the newly added config also goes to there.

To differentiate metrics among multiple clusters, I'd suggest adding a `metrics_prefix` config or `const_labels` config to `ServiceTelemetryMetrics`. In our use case our team are currently using something like `label_replace({__name__=~".*_exporter_sent_spans"}, "cluster", "$1", "__name__", "(.*)_exporter_sent_spans")` to extract the prefix (namespace) into labels to display in grafana, since back then we have no way to add const labels to the metrics (or is there some ways I don't know?) Now that we may be able to add the labels via the yaml directly so adding `const_labels` is enough for our purpose. Or maybe we can add both (i.e. `metrics_prefix` and `const_labels`).

**Comment by bogdandrutu** [comment]
I agree, we are just waiting for the otel-go metrics to be ready, because then we will have "Resource" and resource detectors setup so will have cloud zones/clusters informations.

**Comment by Frefreak** [comment]
That would be great. Do you know the estimated time that will be implemented? My only issue right now is that there seems no way to set the metrics prefix or add custom labels currently.

**Comment by bogdandrutu** [comment]
This is done, @tigrannajaryan  added a hacky "resource".
